### PR TITLE
Fixes #32604 - pin to psych 3.x

### PR DESCRIPTION
**Pull request by lzap**
VCR is configured to use Psych as the serializer. Psych 4.0 changed load to be safe_load. In tests we don't mind, the codebase does not appear to use any YAML loading.

More info:

https://community.theforeman.org/t/psa-rubygem-psych-4-0-is-out-and-its-a-breaking-change/23604

Edit: We need to pin psych gem.

### CODE REVIEW

File: test/models/docker_manifest_list_test.rb
Comment (ezr-ondrej) [inline test/models/docker_manifest_list_test.rb:11]: This seems to be backwards incompatible so the tests are failing there.

Comment (lzap) [reply test/models/docker_manifest_list_test.rb]: Damn so both versions are in use at some point? I guess I need to pin this.

**Comment by lzap** [comment]
Tests are green, we need to unblock CI. Thanks!

**Comment by lzap** [comment]
@ehelms I think the CI is blocked by this.

**Review by ezr-ondrej — APPROVED** [review]
Can we please merge this?
We would like to run katello ci in foreman repo.

**Review by ehelms — APPROVED** [review]

**Comment by lzap** [comment]
For the record, this patch was reverted in https://github.com/Katello/katello/pull/9374

**Pull request merged**
